runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -12821,11 +12826,11 @@ J9::X86::TreeEvaluator::irdbariEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    // For rdbar and wrtbar nodes we first evaluate the children we need to
    // handle the side effects. Then we delegate the evaluation of the remaining
    // children and the load/store operation to the appropriate load/store evaluator.
-   TR::Node *sideEffectNode = node->getFirstChild();
-   TR::Register *sideEffectRegister = cg->evaluate(sideEffectNode);
 
    if (cg->comp()->getOption(TR_EnableFieldWatch))
       {
+      TR::Node *sideEffectNode = node->getFirstChild();
+      TR::Register *sideEffectRegister = cg->evaluate(sideEffectNode);
```

Comment:
What is the reason for moving this evaluation down into this `if` path? Isn't this going to cause a bug?